api/apps/geoprocessing/src/modules/tile/tile.service.ts

```diff
@@ -181,18 +199,51 @@ export class TileService {
    * @todo add tile render type to promise
    */
   async getTile(tileInput: TileInput<string>): Promise<Buffer> {
-    let data: Buffer = Buffer.from('');
     try {
-      const queryResult: Record<
-        'mvt',
-        Buffer
-      >[] = await this.fetchTileFromDatabase(tileInput);
-      // zip data
-      data = await this.zip(queryResult[0].mvt);
+      const {table, z, x, y , customQuery } = tileInput;
+      const filters = !!customQuery ? customQuery : '';
+      const cacheOptions: TileCacheOptions = {
+        enabled: true,
+        redisOptions: {
+          ttl: 86400, // 24 hours
+          host: process.env.REDIS_HOST,
+        },
+      };
+
+      const  redisCache = new Redis(cacheOptions.redisOptions);
+      const cacheKey = `${table}-${z}-${x}-${y}-${filters}`
```

Comment:
This initial block of code is all that we are wrapping in the outermost `try` block - I think we can remove this `try/catch` outright and if anything happens here, let the global exception filter deal with it.

I would even go a bit further and avoid any nested `try/catch` in this function altogether: I think we can get away with something like this (totally untested!)

```typescript
async getTile(tileInput: TileInput<string>): Promise<Buffer> {
    const { table, z, x, y, customQuery } = tileInput;
    const filters = !!customQuery ? customQuery : [];
    const cacheOptions: TileCacheOptions = {
      enabled: true,
      redisOptions: {
        ttl: 86400, // 24 hours
        host: process.env.REDIS_HOST,
      },
    };

    const redisCache = new Redis(cacheOptions.redisOptions);
    const cacheKey = `${table}-${z}-${x}-${y}-${filters}`;

    const tileFromCache = await redisCache
      .getBuffer(cacheKey)
      .catch((error) => {
        // In case the cache get fail, we continue to generate the tile
        this.logger.error(`Cache failed: ${error.message}`);
      });

    if (tileFromCache) return tileFromCache;

    return await this.fetchTileFromDatabase(tileInput)
      .then(async (result) => {
        return await this.zip(result[0].mvt);
      })
      .then(async (result) => {
        redisCache.set(cacheKey, result, 'EX', cacheOptions.redisOptions?.ttl);

        return result;
      })
      .catch((error) => {
        this.logger.error(`Database error: ${error.message}`);
        throw new Error(error.message);
      });
  }
```

Reply:
also, don't forget to run `yarn format` before a final push

Reply:
and `yarn add -D @types/ioredis` would be useful for development too

Reply:
everything done! thanks for your help!

Reply:
@elpamart does this still work? 😅 
I see you've taken up most of my suggestion for this function, but be aware that I had only checked that it made sense from a typing point of view and *very roughly* logic-wise, but I didn't check that it would do what it's supposed to do 😱 

Feel free to merge if you've checked that things work the way they did with your initial implementation, but I just wanted to clarify that I didn't check that my suggested implementation *actually worked* (nor have I done so now on the latest commit in this PR) 😉